sound/soc/sof/intel/hda-stream.c

```diff
@@ -659,13 +659,15 @@ void hda_dsp_stream_free(struct snd_sof_dev *sdev)
 #endif
 
 	list_for_each_entry_safe(s, _s, &bus->stream_list, list) {
+		stream = stream_to_hdac_ext_stream(s);
```

Comment:
should stream be defined conditionally? We don't want to use hdac_ext_stream in all cases